kube-client/src/client/tls.rs

```diff
@@ -53,109 +53,116 @@ pub mod native_tls {
-            let (key, certs) = {
-                let mut pem = Cursor::new(buf);
-                let certs = rustls_pemfile::certs(&mut pem)
-                    .and_then(|certs| {
-                        if certs.is_empty() {
-                            Err(std::io::Error::new(
-                                std::io::ErrorKind::NotFound,
-                                "No X.509 Certificates Found",
-                            ))
-                        } else {
-                            Ok(certs.into_iter().map(rustls::Certificate).collect::<Vec<_>>())
-                        }
-                    })
-                    .map_err(|_| Error::SslError("No valid certificate was found".into()))?;
-                pem.set_position(0);
-
-                // TODO Support EC Private Key to support k3d. Need to convert to PKCS#8 or RSA (PKCS#1).
-                // `openssl pkcs8 -topk8 -nocrypt -in ec.pem -out pkcs8.pem`
-                // https://wiki.openssl.org/index.php/Command_Line_Elliptic_Curve_Operations#EC_Private_Key_File_Formats
-                let mut sk = rustls_pemfile::pkcs8_private_keys(&mut pem)
-                    .and_then(|keys| {
-                        if keys.is_empty() {
-                            Err(std::io::Error::new(
-                                std::io::ErrorKind::NotFound,
-                                "No PKCS8 Key Found",
-                            ))
-                        } else {
-                            Ok(keys.into_iter().map(rustls::PrivateKey).collect::<Vec<_>>())
-                        }
-                    })
-                    .or_else(|_| {
-                        pem.set_position(0);
-                        rustls_pemfile::rsa_private_keys(&mut pem).and_then(|keys| {
-                            if keys.is_empty() {
-                                Err(std::io::Error::new(
-                                    std::io::ErrorKind::NotFound,
-                                    "No RSA Key Found",
-                                ))
-                            } else {
-                                Ok(keys.into_iter().map(rustls::PrivateKey).collect::<Vec<_>>())
-                            }
-                        })
-                    })
-                    .map_err(|_| Error::SslError("No valid private key was found".into()))?;
-
-                if let (Some(sk), false) = (sk.pop(), certs.is_empty()) {
-                    (sk, certs)
-                } else {
-                    return Err(Error::SslError("private key or certificate not found".into()));
-                }
-            };
+    ) -> Result<ClientConfig, Error> {
+        let config_builder = if let Some(certs) = root_certs {
+            ClientConfig::builder()
+                .with_safe_defaults()
+                .with_root_certificates(root_store(certs)?)
+        } else {
+            ClientConfig::builder().with_safe_defaults().with_native_roots()
```

Comment:
Loads the native roots.

@clux Can you try Rancher again?

Reply:
Just checked. It's fixed!